functests/5_latency_testing/latency_testing.go

```diff
@@ -0,0 +1,210 @@
+package __latency_testing
+
+import (
+	//"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	//tool to test
+	OSLAT       = "oslat"
+	CYCLICTEST  = "cyclictest"
+	HWLATDETECT = "hwlatdetect"
+	//Environment variables
+	LATENCY_TEST_DELAY         = "LATENCY_TEST_DELAY"
+	LATENCY_TEST_RUN           = "LATENCY_TEST_RUN"
+	LATENCY_TEST_RUNTIME       = "LATENCY_TEST_RUNTIME"
+	MAXIMUM_LATENCY            = "MAXIMUM_LATENCY"
+	OSLAT_MAXIMUM_LATENCY      = "OSLAT_MAXIMUM_LATENCY"
+	HWLATDETEC_MAXIMUM_LATENCY = "HWLATDETEC_MAXIMUM_LATENCY"
+	CYCLICTEST_MAXIMUM_LATENCY = "CYCLICTEST_MAXIMUM_LATENCY"
+	LATENCY_TEST_CPUS          = "LATENCY_TEST_CPUS"
+	//invalid values error messages
+	UNEXPECTED_ERROR = "Unexpected error"
+	//incorrect values error messages
+	INCORRECT_P1 = "the environment variable"
+	INCORRECT_P2 = "has incorrect value"
+
+	//success message
+	SUCCESS = `SUCCESS.*1 Passed.*0 Failed.*2 Skipped`
+	FAIL    = `FAIL.*0 Passed.*1 Failed.*2 Skipped`
+	//failure messages
+	LATENCY_FAIL = `The current latency .* is bigger than the expected one`
+	SKIP         = `SUCCESS.*0 Passed.*0 Failed.*3 Skipped`
+	//skip messages
+	SKIP_TEST_RUN    = `Skip the latency test, the LATENCY_TEST_RUN set to false`
+	SKIP_MAX_LATENCY = `no maximum latency value provided, skip buckets latency check`
+
+	//hwlatdetec fail message
+	HWLATDETEC_FAIL='Samples exceeding threshold: [^0]'
+	//used values parameters
+	GUARNTEED_LATENCY = "20000"
+)
+
+//Struct to hold each test parameters
+type latencyTest struct {
+	testDelay            string
+	testRun              string
+	testRuntime          string
+	testMaxLatency       string
+	oslatMaxLatency      string
+	cyclictestMaxLatency string
+	hwlatdetecMaxLatency string
+	testCpus             string
+	outputMsgs           []string
+	toolToTest           string
+}
+
+var (
+	INCORRECT_CPUS_NUMBER                = fmt.Sprintf("%s %s %s", INCORRECT_P1, LATENCY_TEST_CPUS, INCORRECT_P2)
```

Comment:
@fromanirh @cynepco3hahue I wonder if we use some errors library anywhere in PAO. Because it might be semantically nicer to define these using something like https://pkg.go.dev/errors#New or https://pkg.go.dev/fmt?utm_source=godoc#Errorf